src/api/spec/controllers/webui/groups/users_controller_spec.rb

```diff
@@ -5,57 +5,49 @@
   let(:user) { create(:confirmed_user) }
   let(:admin) { create(:admin_user) }
 
-  RSpec.shared_examples 'response for non existing user or group' do
-    it 'reports an error' do
-      expect(response).to have_http_status(:not_found)
-      expect(flash[:error]).not_to be_nil
-    end
-  end
-
   describe 'POST create' do
     before do
       login(admin)
     end
 
     context 'when the user exists' do
       subject! do
-        post :create, params: { group_title: group.title, user_login: user.login }, format: :js
+        post :create, params: { group_title: group.title, user_login: user.login }
       end
 
       it 'adds the user to the group' do
         expect(response).to redirect_to(group_show_path(title: group.title))
-        expect(flash[:success]).not_to be_nil
+        expect(flash[:success]).to eq("Added user '#{user}' to group '#{group}'")
         expect(group.users.where(groups_users: { user: user })).to exist
       end
     end
 
     context 'when the user does not exist' do
       subject! do
-        post :create, params: { group_title: group.title, user_login: 'unknown_user' }, format: :js
+        post :create, params: { group_title: group.title, user_login: 'unknown_user' }
       end
 
-      include_examples 'response for non existing user or group'
+      it { expect(flash[:error]).to eq("User 'unknown_user' not found") }
       it { expect(group.users.where(groups_users: { user: user })).not_to exist }
     end
 
     context 'when the group does not exist' do
       subject! do
-        post :create, params: { group_title: 'unknown_group', user_login: user.login }, format: :js
+        post :create, params: { group_title: 'unknown_group', user_login: user.login }
       end
 
-      include_examples 'response for non existing user or group'
+      it { expect(flash[:error]).to eq("Group 'unknown_group' not found") }
     end
 
     context 'when there is an error during user creation' do
       let!(:group_user) { create(:groups_user, group: group, user: user) }
 
       subject! do
-        post :create, params: { group_title: group.title, user_login: user.login }, format: :js
+        post :create, params: { group_title: group.title, user_login: user.login }
       end
 
       it 'reports the error' do
-        expect(response).to have_http_status(:bad_request)
-        expect(flash[:error]).not_to be_nil
+        expect(flash[:error]).to eq("Couldn't add user '#{user}' to group '#{group}': User User already has this group")
```

Comment:
The duplication of `User` in the second part of the message is coming from the controller when we call `errors.full_messages.to_sentence` on a `GroupUser` instance. See the output below, especially `@messages`. Since there's the `:user` key, it's added to the message which already contains `User`.

```
 @base=#<GroupsUser:0x00007f872811b050 group_id: 1, user_id: 5, created_at: nil, email: true, id: nil>,
 @details={:user=>[{:error=>"User already has this group"}]},
 @messages={:user=>["User already has this group"]}>
```